### DISCUSSION TITLE
add uv_bootid()

### DISCUSSION THREAD

**Discussion by calvin2021y** [libuv/libuv#4557] [General]
This is useful for a lot case.

**Comment by saghul** [comment]
Can you provide some examples?

**Comment by calvin2021y** [comment]
for linux:

```sh
cat /proc/sys/kernel/random/boot_id
```
 
a unique id, to let process know if the system reboot or not.

**Reply by saghul** [reply]
Do other platforms have an equivalent api?

**Reply by calvin2021y** [reply]
I am not sure.

for macOS, there is `sysctl kern.boottime`,  windows `Get-WmiObject Win32_OperatingSystem | Select-Object LastBootUpTime`

**Reply by saghul** [reply]
I see. @libuv/collaborators WDWT? I don't see why not!

**Reply by bnoordhuis** [reply]
`kern.boottime` is not the same thing as `/proc/sys/kernel/random/boot_id`. It's the time since system start, like `/proc/uptime` on linux.
